add -nx flag to block execution of all files and scripts

hb9kns opened the pull request:
Using gophernicus on an experimental gopher server with git-only access for users ( gopher://dome.circumlunar.space ), I wanted to have the possibility to suppress processing of executables (gophermaps and scripts), therefore I added an -nx option and corresponding code. Seems to work on my system... Thanks for gophernicus, by the way! ^-^

fosslinux commented:
@hb9kns I'm not a maintainer, but may I ask why you added yourself as an author and changed the codename, for such a trivial change?

fos1 commented:
Not me. I did not do anything to your repository. I have never done anything other than read posts. You are welcome to remove me as anything other than a read only member or completely if you wish. Someone else must be using my email. I do not use fosslinux as a handle. Sorry for any confusion.

Sent from my iPhone

> On Feb 23, 2019, at 10:46 PM, fosslinux <notifications@github.com> wrote:
> 
> @hb9kns I'm not a maintainer, but may I ask why you added yourself as an author and changed the codename, for such a trivial change?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

hb9kns commented:
Sorry guys, if I did anything wrong! I'm not a developer with any collaborative/github experience.

Of course I don't claim authorship on gophernicus, but because I need to run the modified code on my server (for security reasons) and at the same time want to tell others what code is running, I've put my modifs into a separate branch where I added a note about them, so that people know it's not the official source tree but a branch I'm responsible for (especially if it's buggy).

Therefore I'm absolutely fine with parts of my branch picked and added to whatever other branch anybody wants, without my name in it!
Apologies if that's not how it should be done.

fosslinux commented:
@fos1 sorry, I'm a bit confused, what do you mean?

fosslinux commented:
@hb9kns No no, nothing wrong! I see why you've done this. If this got merged in the official branch, then technically you wouldn't be responsible for it and it **would** be part of the official codebase. At any rate, it seems @kimholviala is AWOL. I'm trying to contact him to find out if he still plans to develop this, but otherwise, I plan to fork gophernicus.

kimholviala commented:
Howdy,

Nope, not AWOL, just other stuff in my life like some prison time and parlamental elections + other even bigger things I keep to myself.

Anyway, -nx sounds logical to me, and to be honest I got sick of Linux because of systemd and all the other big changes lately (no ifconfig???) which kind of made me lose intrest. Also no need to fork as you can have control of the original project if you want to - but if you want the gophernicus.org domain act quickly because it's going to expire next month.



- Kim

> On 27 Feb 2019, at 0.00, fosslinux <notifications@github.com> wrote:
> 
> @hb9kns No no, nothing wrong! I see why you've done this. If this got merged in the official branch, then technically you wouldn't be responsible for it and it would be part of the official codebase. At any rate, it seems @kimholviala is AWOL. I'm trying to contact him to find out if he still plans to develop this, but otherwise, I plan to fork gophernicus.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

hb9kns commented:
I can well relate to your systemd feelings... it's the reason I've changed all my hardware to FreeBSD and OpenBSD. However, I'm also running some cheap VPS on Ubuntu (yeah, *really* cheap).
I'd love to keep the gophernicus.org domain up and running for promoting gophernicus!
If somebody else wants do it, I'm fine with that as well. Otherwise, I'd volunteer to pay for the domain and host it somewhere!

fosslinux commented:
@kimholviala very understandable that you don't have time.

@hb9kns I'm happy to work with you to try and continue this project. If @kimholviala would like to add us both as collaborators? I'm happy for you to take the gophernicus.org domain, I wouldn't actuallly be able to pay for it anyway.

kimholviala commented:
Howdy,

Whoever wants to control the gophernicus.org domain contact me directly to kim@holviala.com and I'll give you the transfer key. You can have the domain for free - it has been paid until March 26th this year and after that it's the normal'ish 10-15$ per year depending on the registry.



- Kim



> On 28 Feb 2019, at 9:31, Yargo <notifications@github.com> wrote:
> 
> I can well relate to your systemd feelings... it's the reason I've changed all my hardware to FreeBSD and OpenBSD. However, I'm also running some cheap VPS on Ubuntu (yeah, really cheap).
> I'd love to keep the gophernicus.org domain up and running for promoting gophernicus!
> If somebody else wants do it, I'm fine with that as well. Otherwise, I'd volunteer to pay for the domain and host it somewhere!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kimholviala/gophernicus/pull/6#issuecomment-468167897>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AiOg-iWaRnr0HonfoVjg6JiD_jc7yqQ6ks5vR4W6gaJpZM4bBmdm>.
>

kimholviala commented:
Howdy #2,

TBH I don't use Github for anything, it was just a place to host the code... I'll add you two as collaborators as soon as I figure out how to do that :D

I guess I need your usernames or emails for that, right? I can check that out later today, now I'm a bit busy with real-life stuff.


- Kim


> On 1 Mar 2019, at 3:02, fosslinux <notifications@github.com> wrote:
> 
> @kimholviala <https://github.com/kimholviala> very understandable that you don't have time.
> 
> @hb9kns <https://github.com/hb9kns> I'm happy to work with you to try and continue this project. If @kimholviala <https://github.com/kimholviala> would like to add us both as collaborators? I'm happy for you to take the gophernicus.org domain, I wouldn't actuallly be able to pay for it anyway.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kimholviala/gophernicus/pull/6#issuecomment-468501154>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AiOg-msJOG9YVtLUF5q8AjruoypI8KRlks5vSHwTgaJpZM4bBmdm>.
>

hb9kns commented:
Ok, thanks for the invitation, worked!
About the domain, I'm fine with paying for it, no question, so I'll contact you independently by e-mail.
However, if somebody else has a good reason to control the domain (or doesn't want me doing it), I'm fine with that as well. Up to you!
And good luck for the RL challenges!

vext01 commented:
Yes, I agree that we should be able to disable executable maps.

I had added such a flag when adding openbsd pledge and unveil support to gophernicus. When -nx is enabled, the pledge restrictions can be tighter.

I had a similar flag for disabling user gopher roots.

https://man.openbsd.org/pledge.2
https://man.openbsd.org/unveil

Once this change PR is in, I could post that if there's interest?

fosslinux commented:
hey, @hb9kns it seems that you have been added as a collaborator and author. Would you mind if you could add me too? If you have any concerns re this and you would like to talk privately, please feel free to contact me at fosslinux AT aussies DOT space

kimholviala commented:
Howdy,

I would have added you too but I didn't know your Github username (fosslinux?) as I don't really use Github nor know what to look for...

Also, the domain gophernicus.org is still available until the end of this month and then it will expire. So if either of you wants to control it plz email me directly to kim@holviala.com and I'll give you the transfer key.


- Kim



> On 3 Mar 2019, at 9:15, fosslinux <notifications@github.com> wrote:
> 
> hey, @hb9kns <https://github.com/hb9kns> it seems that you have been added as a collaborator and author. Would you mind if you could add me too? If you have any concerns re this and you would like to talk privately, please feel free to contact me at fosslinux AT aussies DOT space
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kimholviala/gophernicus/pull/6#issuecomment-468996168>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AiOg-orQy7dVZ4fTPq7nEfZz6P1NzxEIks5vS3Z1gaJpZM4bBmdm>.
>

fosslinux commented:
@kimholviala yeah fosslinux is my github username. i think @hb9kns can do it too

hb9kns commented:
> @kimholviala yeah fosslinux is my github username. i think @hb9kns can do it too

I don't see any way how to add another contributor. Maybe I'm blind, but the repo settings are not available to me; I suppose that remains with the repo owner.

kimholviala commented:
I already added fosslinux as a contributor to the repo - or, sent the request which I guess still needs to be accepted.


- Kim

> On 4 Mar 2019, at 11:21, Yargo <notifications@github.com> wrote:
> 
> @kimholviala <https://github.com/kimholviala> yeah fosslinux is my github username. i think @hb9kns <https://github.com/hb9kns> can do it too
> 
> I don't see any way how to add another contributor. Maybe I'm blind, but the repo settings are not available to me; I suppose that remains with the repo owner.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kimholviala/gophernicus/pull/6#issuecomment-469178214>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AiOg-vT_0OO5p4zS7OVQo_nXOitfmqSdks5vTOWggaJpZM4bBmdm>.
>

hb9kns commented:
Thanks for merging! I was asking myself whether it would be good practice to accept and merge one's own pull requests... ^-^

fosslinux commented:
@hb9kns hehe, I ask my self the same question, imo if someone else "approves" it/looks over the code, then it's ok :)

kimholviala commented:
Well, I always just accepted my own changes :D - but you two can decide what you want to do. I might go through some diffs but probably not everything due to lack of time. Anyway, that -nx patch looked ok, but the one thing I started wondering last night was does it then dump the executable as is to the user? Didn't check, just wondering out loud.


- Kim


> On 4 Mar 2019, at 22:48, Yargo <notifications@github.com> wrote:
> 
> Thanks for merging! I was asking myself whether it would be good practice to accept and merge one's own pull requests... ^-^
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/kimholviala/gophernicus/pull/6#issuecomment-469415748>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AiOg-pnltaBa2-sckxEAC7wmaGLzUDMIks5vTYahgaJpZM4bBmdm>.
>

fosslinux commented:
Had to dig into an old email address to find the invite link. Thanks @kimholviala , it worked!
